### ISSUE TITLE
How can I implement this model only RGB and Flow feature

### ISSUE THREAD

**Issue by bmnptnt** [shengyangsun/MSBT#1]
Hi, I'm studying Video Anomaly Detection now.
First, Thank you for your contribution.

I'm wondering that can I implement your model with out audio feature?
What I mean is only RGB and Flow feature.
If it is possible, How can modify you code?

Thank you.

**Comment by shengyangsun** [comment]
Thank you for your interest in our work.

Our MSBT module, as implemented in the `MultiScaleBottleneckTransformer.py` file, intrinsically supports the fusion of two modalities without requiring any modifications. 

First, you need to modify the `load_dataset.py` file so that it only returns RGB and Flow features. 

Next, you need to update the `MultimodalTransformer.py` file to ensure that only RGB and Flow are used during feature fusion. For example：

f_v, f_f = self.fc_v(f_v), self.fc_f(f_f)
f_v, f_f = self.msa(f_v), self.msa(f_f)
f_vf, b_vf = self.MST(f_v, f_f)
f_fv, b_fv = self.MST(f_f, f_v)
bottle_cat = torch.cat([b_vf, b_fv], dim=1)

**Comment by bmnptnt** [comment]
Thank you for your quick and detailed response.

**Comment by ncxa** [comment]
Thanks，two mentors, I want to abandon the flow feature and use RGB and audio features. The code seems to require more than the above changes. Do you have any successful experiences for me to learn from

**Comment by shengyangsun** [comment]
> Thanks，two mentors, I want to abandon the flow feature and use RGB and audio features. The code seems to require more than the above changes. Do you have any successful experiences for me to learn from

Using only RGB and Audio follows a modification similar to using only RGB and Flow, please refer to the explanation above for details. Additionally, don't forget to adjust the calculation method of `loss_infoNCE` for both types of modifications. For instance, when using only RGB and Audio, the calculation of `loss_infoNCE` should be modified as follows:

```python
loss_infoNCE = .0
if seq_len is not None:
    cnt_n = 0
    n_av, n_va = normalize(f_av, f_va)
    n_list = [n_av, n_va]
    for i in range(len(n_list)):
        for j in range(i + 1, len(n_list)):
            cnt_n += 1
            loss_infoNCE += infoNCE(n_list[i], n_list[j], seq_len)
    loss_infoNCE = loss_infoNCE / cnt_n

    f_av, f_va = [bottle_weight[:, i, :].view([-1, 1, 1]) * f
                  for i, f in enumerate([f_av, f_va])]
    f_av = torch.cat([f_av, f_va], dim=-1)
```
